src/models/state.model.ts

```diff
@@ -0,0 +1,10 @@
+import { ExtensionContext } from "vscode";
```

Comment:
Nice.
We can move the `settings.ts` into each separate classes in this folder ? What do you suggest. Can we do this in this PR ?

Reply:
Sure, this will be great for refactoring the code and making it easier for new users to contribute.